src/ripple/app/paths/RippleState.h

```diff
@@ -153,27 +151,24 @@ class RippleState
     getJson(int);
 
 private:
-    std::shared_ptr<SLE const> sle_;
+    uint256 key_;
 
-    bool mViewLowest;
-
-    std::uint32_t mFlags;
+    STAmount const mLowLimit;
+    STAmount const mHighLimit;
 
-    STAmount const& mLowLimit;
-    STAmount const& mHighLimit;
-
-    AccountID const& mLowID;
-    AccountID const& mHighID;
+    STAmount mBalance;
 
     Rate lowQualityIn_;
     Rate lowQualityOut_;
     Rate highQualityIn_;
     Rate highQualityOut_;
```

Comment:
Since you're in here, only two of these four values are used by this class (either the `low` or the `high` versions, depending on `mViewLowest`, so we can save 8 bytes if we remove the other two.

We _could_ actually save 16 bytes by removing all four values entirely; they only place where they're referenced from is in `AccountLines.cpp` at which point, you might as well reload the `SLE` and grab these values directly.

Reply:
fixed in `7a5b4d3665 [fold] Rm rate from trust line cache for the path finder`

I did some refactoring to do this. We know have a base class `TrustLineBase` and two classes that inherit from it `PathFindTrustLine` that doesn't have the qualities, so we save space on the tens of millions of those. We also have an `RPCTrustLine` that has the rates that is used for the `AccountLines` command.